src/components/forms/pool_actions/WithdrawForm/composables/useWithdrawMath.ts

```diff
@@ -489,19 +489,33 @@ export default function useWithdrawMath(
     const tokensIn = amounts.map(() => pool.value.address);
     const fetchPools = !batchSwap.value; // Only needs to be fetched on first call
 
-    const result = await balancer.swaps.queryBatchSwapWithSor({
-      tokensIn: tokensIn,
-      tokensOut: tokensOut || batchSwapTokensOut.value,
-      swapType,
-      amounts,
-      fetchPools: {
-        fetchPools,
-        fetchOnChain: false
+    try {
+      const result = await balancer.swaps.queryBatchSwapWithSor({
+        tokensIn: tokensIn,
+        tokensOut: tokensOut || batchSwapTokensOut.value,
+        swapType,
+        amounts,
+        fetchPools: {
+          fetchPools,
+          fetchOnChain: false
+        }
+      });
+      batchSwapLoading.value = false;
+      return result;
+    } catch (error) {
+      if ((error as Error).message.includes('SOR Swap returned 0 amount')) {
```

Comment:
I think we need to add specific type-safe errors to the SDK so we can handle these safely. At the moment I need to catch the specific error by the error message, but what if the SDK updates and changes this error message?

I have an abstract solution here: https://www.typescriptlang.org/play?#code/PQKhFgCgAIWh1AkgFQBKIHLQMoBEDSUswUApgHYCuAttAEICGANg+QMakBOAop5wPacAwvwAmpaAG8iOeAEEACgH0AWtwBKAeSXruyAKrqMSuQFlN+jMmgBeaAHJs85Wq069h42YtX7UAL5QUGwsAM6h9MysHDx8gtCkAB4ALhSiEbwCnFIybPzkocmclGzJggAUAA6UAEZMAJZs0HniAFyRLOxcmYIi4gCUOTDQI6GUlVzljJ0xPZwAdADmpMmmpOEMy+UtpP39ANwyI8kAFvWh8+QM1BJ2AETT0d1xnHeHw4GQMoUMyY3Qy1W61Cm1I2zEpHajy6sSyfV2QxGowA7vVkmwTtBwQNEUiRmwGKEJNDZi94fMnIpVBptLoDEYTOZLMhWkc8dBOCtKJxyA5sJp1DhkQxKhyuTzSKJoAAGaDXfiUcjJezvdnQcQAMwYlCYyVZwzVnOS3N59n05AA1uR+MjeVwsiq2Z8Rp9PlBgMAcAQEolKvwiVKNYrSvV8lAg+w-vloKF+DdsKILQAxYNR8jlQbSYYe6C4fgx40ajVy8hS+3xc7QU4CW0yas26DkUjIjpPWEVEnPOEQinOaluOmeRk+ZAHALusBEODcDC4BAodBYPCEGAgEhfDcTiCr+doTDQISaUzYfSmDTQRQKAAyiCEcmQiE0GCn65zIgKNC4csqlQaBLTjakJKETCkqERlM0vwYiWUonKwohMBIoQTGw9Qav85acOGqahrysY3O+Yw3JwKaRrhGa4kUACeuKjHGpAJsmOH5BmqrQP4kHopi5SYZmbJoVimHQPUBTJE8-DFp27bZAAZDJCQvPMOy2DYdhSXM5KUi4NLuPSXhMlYfEGkiOYYA2yISASvJwaWiFVmcETIaQqHoU0QmoqcVZURMMYMBqKxUWy+L5LGiHzEw-CLOU9hjGwHDhEGTBMDRNkIUhKFoRhLz2P0ToJEwRK0Ui9YtphbEujIrpBJA+GkIRn4kcx6ZjpAQA

Lets add something like this to the SDK.